### PR TITLE
correct needsUpdate when strlen (or count) = 1 which incorrectly equa…

### PR DESCRIPTION
**Pull request by Ruud68**
…led to 'true'

Pull Request for Issue # .

### CODE REVIEW

**Comment by laoneo** [comment]
The code ? false : true is not needed

**Comment by Ruud68** [comment]
it is :) when omitted values of len 1 will NOT be saved see screencapture: https://office.onlinecommunityhub.nl/nextcloud/s/MqxyoWang8orToQ

File: administrator/components/com_fields/models/field.php [hunk 1]
```diff
@@ -590,7 +590,7 @@ public function setFieldValue($fieldId, $itemId, $value)
 		elseif (count($value) == 1 && count((array) $oldValue) == 1)
 		{
 			// Only a single row value update can be done when not empty
-			$needsUpdate = is_array($value[0]) ? !count($value[0]) : !strlen($value[0]);
+			$needsUpdate = is_array($value[0]) ? count($value[0]) === 0 ? false : true : strlen($value[0]) === 0 ? false : true;
```
Comment (laoneo) [inline administrator/components/com_fields/models/field.php]: What I mean is, does the following code the same?
`$needsUpdate = is_array($value[0]) ? count($value[0]) !== 0 : strlen($value[0]) !== 0;`

Comment (Quy) [reply administrator/components/com_fields/models/field.php]: Yes, it works and shorter/more readable.

Comment (Ruud68) [reply administrator/components/com_fields/models/field.php]: top, works! Way better then my fiddle :)

**Pull request merged**